chalice/deploy/planner.py

```diff
@@ -279,14 +290,100 @@ def plan_scheduledevent(self, resource):
         ]
         return plan
 
+    def plan_restapi(self, resource):
+        # type: (models.RestAPI) -> List[models.Instruction]
+        function = resource.lambda_function
+        function_name = function.function_name
+        varname = '%s_lambda_arn' % function.resource_name
+        lambda_arn_var = Variable(varname)
+        shared_plan = [
+            # The various API gateway API calls need
+            # to know the region name and account id so
+            # we'll take care of that up front and store
+            # them in variables.
+            models.BuiltinFunction(
+                'parse_arn',
+                [lambda_arn_var],
+            ),
+            models.StoreValue('parsed_lambda_arn'),
+            models.JPSearch('account_id'),
+            models.StoreValue('account_id'),
+            models.LoadValue('parsed_lambda_arn'),
+            models.JPSearch('region'),
+            models.StoreValue('region_name'),
+            # The swagger doc uses the 'api_handler_lambda_arn'
+            # var name so we need to make sure we populate this variable
+            # before importing the rest API.
+            models.LoadValue(varname),
+            models.StoreValue('api_handler_lambda_arn'),
```

Comment:
This `shared_plan` is a little hard for me to follow. Is this just taking all of the variables from the parse_arn and storing it as variables to use later? We talked about this offline but I realize that this is suppose to be really low level, but I think in the future it would be nice if we moved over to a register-based implementation. Keeping track of where the stack is at is a little difficult as you have to know how each opcode affects the stack.

Reply:
>  Is this just taking all of the variables from the parse_arn and storing it as variables to use later?

Yes.   This is needed for both the create as well as the update case, so it's a shared set of instructions.